libbeat/cfgfile/reload.go

```diff
@@ -174,29 +175,34 @@ func (rl *Reloader) Run(runnerFactory RunnerFactory) {
 					continue
 				}
 
-				runner, err := runnerFactory.Create(c)
+				rawCfg := map[string]interface{}{}
```

Comment:
Not 100% sure anymore, but I think there is a difference between this and `ID()`. In the case of `ID()` the following to configs are the same:

```
a: b
c: d
```

and

```
c: d
a: b
```

That means the order does not matter, but with the above it does. But not 100% sure anymore if that was the reason I ended up with `ID()` but there was an issue.

Reply:
```
func main () {
	a := map[string]interface{}{
		"a": "b",
		"c": "d",
	}


	b := map[string]interface{}{
		"c": "d",
		"a": "b",
	}

	fmt.Println(hashstructure.Hash(a, nil))
	fmt.Println(hashstructure.Hash(b, nil))
}

```

output:
```
8710980024335696864 <nil>
8710980024335696864 <nil>

```